src/GameWorldSimulator/NeoServer.Game.Creatures/Models/Bases/CombatActor.cs

```diff
@@ -437,23 +438,37 @@ protected void ReduceHealth(CombatDamage damage)
 
     public abstract ILoot DropLoot();
 
-    public virtual void OnDeath(IThing by)
+    public virtual void Death(IThing by)
     {
+        if (by is ICombatActor combatActor)
+        {
+            //todo: implements real damage
+            OnPrepareDeath?.Invoke(this, combatActor, 0);
+            combatActor.Kill(this);
```

Comment:
the victim object is calling the enemy "Kill" method which is not responsibility of the method. let's think on another solution

Reply:
ok, this is a example of use:
![image](https://github.com/user-attachments/assets/fed88741-88b5-4f68-af41-44e7fb873a0a)